File: scala/rna-transcription/src/main/scala/RnaTranscription.scala
import javax.print.CancelablePrintJob

object RnaTranscription {
  def toRna(dna: String): Option[String] = {
    def translate(char: Char): Char =
      char match {
        case 'A' => 'U'
        case 'C' => 'G'
        case 'G' => 'C'
        case 'T' => 'A'
      }

    if (dna.isEmpty) None else Some(dna.map(translate))
  }
}

File: elixir/rna-transcription/lib/rna_transcription.ex
defmodule RnaTranscription do
  @dna_rna_mapping %{
    ?A => ?U,
    ?C => ?G,
    ?T => ?A,
    ?G => ?C
  }

  @doc """
  Transcribes a character list representing DNA nucleotides to RNA


  ## Examples

  iex> RnaTranscription.to_rna('ACTG')
  'UGAC'
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    dna
    |> Enum.map(& @dna_rna_mapping[&1])
  end
end
